Fix for issue #83

Chopknee opened the pull request:
Changed raw key access to a get with safe default when dealing with IMDB returned data.

Commit:
Changed raw key access to a get with safe default when dealing with IMDB returned data.

squirrelo commented on movienightbot/db/controllers.py:
```diff
@@ -102,13 +102,14 @@ def update_imdb_id(self, server_id: int, movie_name: str, imdb_id: str):
-            "year": imdb_info["year"],
-            "thumbnail_poster_url": imdb_info["cover url"],
-            "full_size_poster_url": imdb_info["full-size cover url"],
+            "canonical_title": imdb_info.get("canonical title", "N/A"),
```
this should probably default to `imdb_info["title"]` for ease of use

squirrelo commented on setup.py:
```diff
@@ -3,11 +3,12 @@
 
 requirements = [
-    "py-cord[speed]>=2.0",
+    "py-cord[speed]==2.4.0",
```
why the pin and not `>=2.4`?

squirrelo commented on movienightbot/commands/suggest.py:
```diff
@@ -40,10 +40,10 @@ def imdb_data(movie: str, kind: str) -> Tuple[Union[None, IMDBInfo], Union[None,
-        "year": imdb_info["year"],
-        "thumbnail_poster_url": imdb_info["cover url"],
-        "full_size_poster_url": imdb_info["full-size cover url"],
+        "canonical_title": imdb_info.get("canonical title", "N/A"),
```
same here

squirrelo replied:
This still needs to be title for default canonical title value

Chopknee committed:
Update setup.py

Reverted py-cord version number
Changed discord.py to >= version number

Chopknee commented on setup.py:
```diff
@@ -8,6 +8,7 @@
     "marshmallow",
     "pyyaml",
     "cinemagoer>=2022.12.27",
+    "discord.py>=2.3.2"
```
If this package isn't installed, the bot errors out and does not start. It is not installed as a dependency of other packages.

Chopknee committed:
Update controllers.py

cononical_title will set to the title if the value isn't provided.

thumbnail_poster_url will set to the full_size_poster_url if none is provided. It will be empty if no full_size_poster_url is not provided either. 
The full_size_poster_url is set in the same way.

Chopknee reviewed (commented):
Addressed the issues brought up.

Sorry, I'm mostly unfamiliar with using github. :)

Chopknee committed:
Update setup.py

Reverted script back to it's original state

Chopknee committed:
Update setup.py

Will eventually get this back to actual original state.

Chopknee committed:
Merge branch 'develop' into missing_cover_83

squirrelo commented on movienightbot/db/controllers.py:
```diff
@@ -102,13 +102,14 @@ def update_imdb_id(self, server_id: int, movie_name: str, imdb_id: str):
-            "full_size_poster_url": imdb_info["full-size cover url"],
+            "canonical_title": imdb_info.get("canonical title", imdb_info["title"]),
+            "year": imdb_info.get("year", 0),
+            "thumbnail_poster_url": imdb_info.get("cover url", imdb_info.get("full-size cover url", "")),
```
This flipped full size into thumbnail and vice versa

Chopknee committed:
Update suggest.py

canonical_title now defaults to title if not provided

Chopknee committed:
Update controllers.py

Keeping it simple. If the poster URL isn't provided, then the field will be left blank.
